python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -674,40 +675,68 @@ def unmanaged_workers(self):
-        tmp += self.info_string(nodes)
-        tmp += "\n"
-        tmp += self.load_metrics.info_string()
-        tmp += "\n"
-        tmp += self.resource_demand_scheduler.debug_string(
-            nodes, self.pending_launches.breakdown(),
-            self.load_metrics.get_resource_utilization())
-        if _internal_kv_initialized():
-            _internal_kv_put(DEBUG_AUTOSCALING_STATUS, tmp, overwrite=True)
-        logger.debug(tmp)
-
-    def info_string(self, nodes):
-        suffix = ""
-        if self.updaters:
-            suffix += " ({} updating)".format(len(self.updaters))
-        if self.num_failed_updates:
-            suffix += " ({} failed to update)".format(
-                len(self.num_failed_updates))
-
-        return "{} nodes{}".format(len(nodes), suffix)
-
-    def request_resources(self, resources: List[dict]):
-        """Called by monitor to request resources.
+    def summary(self):
+        """Summarizes the active, pending, and failed node launches.
 
-        Args:
-            resources: A list of resource bundles.
+        An active node is a node whose raylet is actively reporting heartbeats.
+
+        A pending node is non-active node whose node tag is uninitialized,
+        waiting for ssh, syncing files, or setting up.
+
+        If a node is not pending or active, it is failed.
+
+        Returns:
+            (AutoscalerSummary) The summary.
         """
-        if resources:
-            logger.info(
-                "StandardAutoscaler: resource_requests={}".format(resources))
-        assert isinstance(resources, list), resources
-        self.resource_demand_vector = resources
+        all_node_ids = self.provider.non_terminated_nodes(tag_filters={})
+
+        active_nodes = Counter()
+        pending_nodes = []
+        failed_nodes = []
+
+        for node_id in all_node_ids:
+            # TODO (Alex): Verify the head node case before merging this.
+            ip = self.provider.internal_ip(node_id)
+            node_tags = self.provider.node_tags(node_id)
+            if node_tags[TAG_RAY_NODE_KIND] == NODE_KIND_UNMANAGED:
+                continue
+            node_type = node_tags[TAG_RAY_USER_NODE_TYPE]
+
+            # TODO (Alex): If a node's raylet has died, it shouldn't be marked
+            # as active.
+            is_active = self.load_metrics.is_active(ip)
+            if is_active:
+                active_nodes[node_type] += 1
+            else:
+                print(f"{ip}: {node_tags}")
+                status = node_tags[TAG_RAY_NODE_STATUS]
+                pending_states = [
+                    STATUS_UNINITIALIZED, STATUS_WAITING_FOR_SSH,
+                    STATUS_SYNCING_FILES, STATUS_SETTING_UP
+                ]
+                is_pending = status in pending_states
+                if is_pending:
+                    pending_nodes.append((ip, node_type))
+                else:
+                    failed_nodes.append((ip, node_type))
```

Comment:
This is problematic. Autoscaler automatically terminates failed_nodes. You probably want to get these nodes from where they are terminated as I marked above.

Reply:
In general, shouldn't we make the pending nodes as we do inside resource demand scheduler?  the delta between non_terminated_node() and load_metric's ips? 
In that case you should basically have:
```
            is_active = self.load_metrics.is_active(ip)
            if is_active:
                active_nodes[node_type] += 1
            else:
                pending_nodes.append((ip, node_type))
```

Reply:
hmmm i thought our invariant was that something must be in load metrics to be considered active.

Reply:
Are you ok with the change I suggested?